Mathlib/GroupTheory/GroupAction/FixedPoints.lean

```diff
@@ -0,0 +1,289 @@
+      exact a_moved
+  · rw [not_mem_movedBy_iff_mem_fixedBy, mem_fixedBy] at a_fixed
+    rw [a_fixed]
+
+@[to_additive]
+theorem smul_mem_of_movedBy_subset {a : α} {s : Set α} {g : G} (superset : movedBy α g ⊆ s) :
+    g • a ∈ s ↔ a ∈ s := (zpow_one g) ▸ smul_pow_mem_of_movedBy_subset superset 1
+
+@[to_additive]
+theorem smul_pow_preimage_eq_of_movedBy_subset {s : Set α} {g : G} (superset : movedBy α g ⊆ s)
+    (j : ℤ) : (fun a => g^j • a) ⁻¹' s = s := by
+  ext a
+  rw [Set.mem_preimage, smul_pow_mem_of_movedBy_subset superset]
+
+/-- If `movedBy α g ⊆ t`, then `g` cannot send a subset of `t` outside of `t` -/
+@[to_additive]
+theorem smul_pow_subset_of_movedBy_subset {s t : Set α} {g : G}  (t_superset : movedBy α g ⊆ t)
+    (s_ss_t : s ⊆ t) (j : ℤ): (fun a => g^j • a) ⁻¹' s ⊆ t := by
+  rw [<-smul_pow_preimage_eq_of_movedBy_subset t_superset j]
+  repeat rw [Set.preimage_smul]
+  exact Set.smul_set_mono s_ss_t
+
+/-- If `g` and `h` commute, then `g` fixes `h • x` iff `g` fixes `x`. -/
+@[to_additive]
+theorem smul_pow_mem_fixedBy_of_commute {g h : G} (comm : Commute g h) (x : α) (j : ℤ):
+    x ∈ fixedBy α g ↔ h^j • x ∈ fixedBy α g := by
+  suffices ∀ x : α, ∀ h : G, Commute g h → x ∈ fixedBy α g → h^j • x ∈ fixedBy α g by
+    refine ⟨this x h comm, fun hx_in_fixedBy => ?x_in_fixedBy⟩
+    have h₁ : x = h⁻¹^j • h^j • x := by rw [<-mul_smul, inv_zpow', zpow_neg, mul_left_inv, one_smul]
+    rw [h₁]
+    exact this _ _ comm.inv_right hx_in_fixedBy
+  intro x h comm x_in_fixedBy
+  rw [mem_fixedBy, <-mul_smul]
+  rw [Commute.zpow_right comm j]
+  rw [mul_smul, smul_left_cancel_iff]
+  exact x_in_fixedBy
+
+/-- If `g` and `h` commute, then `g` moves `h • x` iff `g` moves `x`. -/
+@[to_additive]
+theorem smul_pow_mem_movedBy_of_commute {g h : G} (comm : Commute g h) (a : α) (j : ℤ):
+    a ∈ movedBy α g ↔ h^j • a ∈ movedBy α g := by
+  repeat rw [movedBy_eq_compl_fixedBy]
+  repeat rw [Set.mem_compl_iff]
+  rw [smul_pow_mem_fixedBy_of_commute comm]
+
+@[to_additive]
+theorem movedBy_eq_smul_pow_movedBy_of_commute {g h : G} (comm : Commute g h) (j : ℤ):
+    movedBy α g = (fun a => h^j • a) ⁻¹' movedBy α g := by
+  ext a
+  rw [Set.mem_preimage, smul_pow_mem_movedBy_of_commute comm]
+
+@[to_additive]
+theorem movedBy_eq_smul_movedBy_of_commute {g h : G} (comm : Commute g h):
+    movedBy α g = (fun a => h • a) ⁻¹' movedBy α g := by
+  nth_rw 1 [movedBy_eq_smul_pow_movedBy_of_commute comm 1]
+  rw [zpow_one]
+
+end Image
+
+section Faithful
+
+variable [FaithfulSMul G α]
+variable [FaithfulSMul M α]
+
+/-- If the action is faithful, then an empty `movedBy` set implies that `m = 1` -/
+@[to_additive]
+theorem movedBy_empty_iff_eq_one {m : M} : movedBy α m = ∅ ↔ m = 1 := ⟨
+  (by
+    intro moved_empty
+    apply FaithfulSMul.eq_of_smul_eq_smul (α := α)
+    intro a
+    rw [one_smul]
+    by_contra ma_ne_a
+    rwa [<-ne_eq, <-mem_movedBy, moved_empty] at ma_ne_a
+  ),
+  fun eq_one => eq_one.symm ▸ movedBy_one_eq_empty α M
+⟩
```

Comment:
This closing bracket should move to the previous line.
Alternatively, consider starting the tactic proof (`by`-block) on line 263, and splitting into two parts with `constructor`.

Reply:
I did the latter, since most of the proof was in a `by`-block anyway